var/spack/repos/builtin/packages/py-mlflow/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyMlflow(PythonPackage):
+    """MLflow: A Platform for ML Development and Productionization."""
+
+    homepage = "https://pypi.org/project/mlflow/"
+    pypi = "mlflow/mlflow-2.0.1.tar.gz"
+
+    version("2.0.1", sha256="7ce6caf3c6acb022d6f5ce8a0995a92be1db524ae16aade1f83da661cdf993de")
+    version("1.17.0", sha256="4898c58899e3101e09e2b37cf5bee7db04c5d73389a56942d3ef5a5e4396799e")
+
+    variant(
+        "skinny",
+        default=False,
+        description="Lightweight MLflow package without SQL storage, server, UI, "
+        "or data science dependencies.",
+    )
```

Comment:
I think we should actually remove this variant, and here's why.

I usually try to emulate pip as best I can so that installing Python packages with Spack is no different than installing them with pip. Pip has a concept of "extras" which is basically the same as our "variants", so it's not uncommon to add variants to match those. But "skinny" isn't an extra, it's an entirely different package: https://pypi.org/project/mlflow-skinny/. The source code for both is the same, but you don't `pip install mlflow[skinny]`, you `pip install mlflow-skinny`. So I think if you want `mlflow-skinny`, there should be a new `py-mlflow-skinny` package for it.

Although `mlflow-skinny` only requires the skinny requirements, `mlflow` proper requires both core and skinny requirements. So you can remove the variant but keep all the deps.

Reply:
Got it - thanks for the clear explanation!